compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/AddContinuationLowering.kt

```diff
@@ -194,7 +194,7 @@ private class AddContinuationLowering(private val context: JvmBackendContext) :
 
                     override fun visitReturn(expression: IrReturn): IrExpression {
                         val ret = super.visitReturn(expression) as IrReturn
-                        return IrReturnImpl(ret.startOffset, ret.endOffset, context.irBuiltIns.anyType, function.symbol, ret.value)
+                        return IrReturnImpl(ret.startOffset, ret.endOffset, ret.type, function.symbol, ret.value)
```

Comment:
Thank you for the catch! This, in fact, should be `anyNType`, since `invokeSuspend` returns `ret.type | COROUTINE_SUSPENDED` and `ret.type` can be nullable. Thus, `ret.type` is also incorrect.

Reply:
The type here actually refers to the type of the `IrReturn` as an expression and is always `Nothing` (compare with `IrBuilderWithScope.irReturn`). I was confused about this as well, but it's probably a parameter since even builtin types are deserialized and not available as constants.